batch_inference_v2/batch_inference_v2.py

```diff
@@ -158,15 +157,19 @@ def infer(
                                                     endpoint.
     :param model_endpoint_drift_threshold:          The threshold of which to mark drifts. Defaulted to 0.7.
     :param model_endpoint_possible_drift_threshold: The threshold of which to mark possible drifts. Defaulted to 0.5.
-    :param model_endpoint_sample_set:               A sample dataset to give to compare the inputs in the drift analysis. The default
-                                                    chosen sample set will always be the one who is set in the model artifact itself.
+    :param model_endpoint_sample_set:               A sample dataset to give to compare the inputs in the drift analysis.
+                                                    Can be provided as an input or as a parameter. The default chosen sample
+                                                    set will always be the one who is set in the model artifact itself.
 
     raises MLRunInvalidArgumentError: if both `model_path` and `endpoint_id` are not provided
     """
 
     # Loading the model:
     context.logger.info(f"Loading model...")
+    if isinstance(model_path, mlrun.DataItem):
+        model_path = model_path.artifact_url
```

Comment:
just making sure - record_results() knows how to handle artifact_url?

Reply:
yes, well this artifact url is basically the model store path (as a string).